src/connector/src/sink/doris_starrocks_connector.rs

```diff
@@ -187,33 +186,27 @@ impl InserterInnerBuilder {
         })
     }
 
-    // TODO: use hyper 1 or reqwest 0.12.2
-    fn build_request_and_client(
-        &self,
-        uri: String,
-    ) -> (Builder, Client<HttpsConnector<HttpConnector>>) {
-        let mut builder = Request::put(uri);
-        for (k, v) in &self.header {
-            builder = builder.header(k, v);
-        }
-
-        let connector = HttpsConnector::new();
+    fn build_request(&self, uri: String) -> RequestBuilder {
         let client = Client::builder()
             .pool_idle_timeout(POOL_IDLE_TIMEOUT)
-            .build(connector);
+            .redirect(redirect::Policy::none()) // we handle redirect by ourselves
+            .build()
+            .unwrap();
 
-        (builder, client)
+        let mut builder = client.put(uri);
+        for (k, v) in &self.header {
+            builder = builder.header(k, v);
+        }
+        builder
     }
 
     pub async fn build(&self) -> Result<InserterInner> {
-        let (builder, client) = self.build_request_and_client(self.url.clone());
-        let request_get_url = builder
-            .body(Body::empty())
-            .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
-        let resp = client
-            .request(request_get_url)
+        let builder = self.build_request(self.url.clone());
+        let resp = builder
+            .send()
             .await
             .map_err(|err| SinkError::DorisStarrocksConnect(err.into()))?;
+        // TODO: shall we let `reqwest` handle the redirect?
```

Comment:
Sounds good, I wonder if the logic of the following special verdict is an obstacle cc @xuefengze